lib/pure/asynchttpserver.nim

```diff
@@ -300,9 +304,14 @@ proc processRequest(
           break
 
         # Read bytesToRead and add to body
-        # Note we add +2 because the line must be terminated by \r\n
-        let chunk = await client.recv(bytesToRead + 2)
-        request.body = request.body & chunk
+        let chunk = await client.recv(bytesToRead)
+        request.body.add(chunk)
+        # Skip \r\n (chunk terminating bytes per spec)
+        when compileOption("assertions"):
```

Comment:
You don't need this, `assert` is already removed by the compiler when this option is not specified.

Reply:
I was told in a different comment (or maybe IRC) that doing something like this would save an extra string allocation. Additionally, if one compiles with asserts off, will
```nim
        let separator = await client.recv(2)
        assert separator == "\r\n"
```
cause an `Unused identifier` warning?

That said, I don't mind removing it as it definitely looks cleaner without it.